src/rust/src/declarative_asn1/encode.rs

```diff
@@ -87,6 +89,29 @@ impl asn1::Asn1Writable for AnnotatedTypeObject<'_> {
                         .ok_or(asn1::WriteError::AllocationError)?;
                 write_value(writer, &printable_string)
             }
+            Type::UtcTime() => {
+                let val: &pyo3::Bound<'_, UtcTime> = value
+                    .downcast()
+                    .map_err(|_| asn1::WriteError::AllocationError)?;
+                let (datetime, _) =
+                    crate::x509::py_to_datetime_with_microseconds(py, val.get().inner.bind(py))
+                        .map_err(|_| asn1::WriteError::AllocationError)?;
```

Comment:
Why are we using the microsecond variant if we're going to ignore microseconds?

Reply:
fixed